src/features/machine/PolarMachine.js

```diff
@@ -88,13 +88,13 @@ export default class PolarMachine extends Machine {
   }
 
   // Returns whether a given path lies on the perimeter of the circle.
-  onPerimeter(v1, v2, delta=.001) {
+  onPerimeter(v1, v2, delta=1) {
```

Comment:
This 1 is a much bigger delta. Does it need to just be bigger than the precision? Maybe 0.01 or 0.005 would be better?